unpack_action.go

```diff
@@ -12,12 +12,23 @@ type UnpackAction struct {
 	File        string
 }
 
+func UnpackTarArchive(infile, destination, compression string) error {
+
+	unpackTarOpt := map[string]string{
+		"gz":    "z",
+		"bzip2": "j",
+		"xz":    "J",
+	} // Trying to guess all other supported formats
+
+	os.MkdirAll(destination, 0755)
+	log.Printf("Unpacking %s\n", infile)
+
+	return Command{}.Run("unpack", "tar", "-x"+unpackTarOpt[compression], "-f", infile, "-C", destination)
```

Comment:
I'm not too fond if having it optional though, as that relies on what the system tar does. But while that's reasonable it still means that if a compression is given ti should be verified if it's one on the known list